### PR TITLE
[5.5] Improved error handling consistency for unhandled HTTP status codes

### PR DESCRIPTION
**Pull request by GuidoHendriks**
Based on the discussion in PR https://github.com/laravel/framework/pull/22820, I decided to create a new PR to solve a slightly different problem.

When handing `HttpException`s there's the possibility to create custom views for any status. By default there are views for the status codes 404, 419, 429, 500 and 503. When in debug other statuses will either use `Whoops` or `SymfonyExceptionHandler` as a fallback, which makes sense.

If not in debug it will currently fallback to `SymfonyExceptionHandler`. This PR changes this to fallback to a separate view called `unhandled.blade.php`, which uses the same layout as predefined views (`{status}.blade.php`). This way all errors are shown consistently in production and could easily be customised to match the application's layout.

I noticed that any exception that's not a `HTTPException` are handled in a similar way: In debug it's shown by `Whoops` or `SymfonyExceptionHandler`, otherwise it's handled as a HTTP 500 which uses `500.blade.php`.

It didn't make sense to also use `500.blade.php` for this, that's why I went with `unhandled.blade.php`. In the other PR the suggested name was `default.blade.php` which could also work, but this is less likely to cause confusion in my opinion.

In short, when handling HTTP status codes that don't have an error view, the shown error will in debug change from:

screen shot 2018-01-17 at 02 10 19

To:

screen shot 2018-01-17 at 13 20 33

And will be customisable like all other HTTP errors.

### CODE REVIEW

**Comment by deleugpn** [comment]
Heads up on merge conflict between both PRs.

**Comment by taylorotwell** [comment]
I dunno. I guess I'm just not seeing the huge benefit here? It seems like you know what status codes your app could respond with, and can make the pages for them, even if they just all extend some basic error layout (which is already supported).

**Comment by GuidoHendriks** [comment]
Do things have to be a "huge" benefit to be considered useful? This would make error handling more consistent, as there are currently multiple ways that unhandled errors are dealt with. Not a huge deal, but quite sloppy, in my opinion. Especially if you know about it.
